docs/commerce/cart/cart.md

```diff
@@ -0,0 +1,100 @@
+---
+description: The Cart component covers adding items to the shopping cart, as well as previewing and modifying the cart information.
+edition: commerce
+---
+
+# Cart
+
+The Cart component is a foundation of the Commerce offering delivered as part 
+of [[= product_name =]].
+It covers actions related to the creation and handling of a list of products 
+that the buyer intends to purchase.
+
+
+The Cart component exposes the following:
+
+- [PHP API](cart_api.md) that allows for managing carts and cart entries, or validating products
+- [REST API](../../api/rest_api/rest_api_reference/rest_api_reference.html#managing-ecommerce-carts) that helps get cart and products information over HTTP
+- [Twig functions](cart_twig_functions.md) that enable checking whether product can be added to cart and formatting the price
+
+There is no specific configuration related to the Cart component.
+All configuration is done at the Checkout and Storefront level.
+
+## Cart data handling
+
+Cart data is handled by two storages, depending on whether the buyer is anonymous 
+or has been authenticated.
+Information that relates to anonymous users is stored in the PHP session, while 
+registered user data is stored in a database.
+
+By default, anonymous users can add items to cart, but to display the cart view, 
+they have to log in and transition into an authenticated user.
+
+!!! note 
+
+    For information about roles and permissions that control access to the cart, 
+    see [Permission use cases](../../permissions/permission_use_cases.md#commerce).
+
+### Cart data merging
+
+When a buyer browses the storefront anonymously and fills the cart with items, 
+anonymous cart data is stored in the PHP session storage.
+Then, when an anonymous user logs into the storefront, cart data from the PHP session 
+storage is persisted and merged with any cart information that might already exist in the database for this authenticated user.
+
+If no previous cart data exists, a new cart is created.
+
+### Cart data validation
+
+When a buyer tries to add products to the cart, increase cart item quantity or proceed to checkout, the Cart component performs cart item validation and checks whether:
+
+- the product is available 
+- the requested quantity of product is available 
+- the product is available at a price in the currency selected for the cart 
+
+## Front-end perspective
+
+From the front-end perspective, the cart consists of a main `Cart` object 
+and several widgets.
+`Cart` is a standalone JavaScript object that manages cart data and has no user interface, 
+while widgets consist of JavaScript code and accompanying Twig templates.
+
+### Cart object
+
+The `Cart` object stores cart entry data and a cart summary, which contains additional entry data, like, for example, formatted gross price or validation errors.
```

Comment:
Not true. Gross price and preformatted price is only held by `CartSummary` object and it's intended to be used by REST API. Normal Cart object has:
- numerical ID
- UUID
- name
- owner
- currency
- entries
- creation date
- modification date

Reply:
We discussed this in private and the original statement is not wrong, but may be confusing, because we have `Cart` object in the REST, which is sent from the backend to the frontend and we have `Cart` class, which instance can also be called `Cart` object and which acts as a cart service, but it is not a cart service in our frontend terminology. And these two objects are different carts.

We need to somehow, call `Cart` "service" object in a way that it won't be confused with `Cart` object from the REST.